mail_activity_team/models/calendar_event.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+
+
+class CalendarEvent(models.Model):
+    _inherit = 'calendar.event'
+
+    def _get_default_team_id(self, user_id=None):
+        if not user_id:
+            user_id = self.env.uid
+        res_model = self.env.context.get('default_res_model')
+        model = self.env['ir.model'].search(
+            [('model', '=', res_model)], limit=1)
+        domain = [('member_ids', 'in', [user_id])]
+        if res_model:
+            domain.extend(['|', ('res_model_ids', '=', False),
+                           ('res_model_ids', 'in', model.ids)])
+        return self.env['mail.activity.team'].search(domain, limit=1)
+
+    privacy = fields.Selection(selection_add=[
+        ('team', 'Only team'),
+    ])
+    team_id = fields.Many2one(
+        comodel_name='mail.activity.team',
+        default=lambda s: s._get_default_team_id(),
+    )
+
+    @api.multi
+    def read(self, fields=None, load='_classic_read'):
```

Comment:
any chance to improve this?

Eg:

0. comment on top what you are doing
1. delegate fields to be used to another method
2. do not browse for each result, collect ids instead 
3. make what happens in the loop clear
4. recurrent_fields and public_fields are static, do not set them at every loop

Reply:
Made some of these changes, I dont quite understand what you mean in number 2.

Reply:
```py
for rec_id in rec_ids:
    rec = self.browse(rec_id)
```
can be more expensive than
```py
self.browse(rec_ids)
```

Reply:
Do you mean this line `team = self.env['mail.activity.team'].browse(team_id)` ?
I dont know if in this case it is worth it to get all possible teams in one browse and then filter them to get one `team_id` in every loop.